sessionctx/variable/session.go

```diff
@@ -1328,6 +1328,12 @@ type SessionVars struct {
 
 	// Resource group name
 	ResourceGroupName string
+
+	// protectedTSList holds a list of timestamps that should delay GC.
+	protectedTSList struct {
```

Comment:
Is it possible to store `protectedTSList` in `ProcessInfo`, so the calculation source of GC safepoint always comes from the session `ProcessInfo` struct.

Reply:
The protected ts list has the same lifetime as the session, however there are multiple processes during that session. So, we may let each process info within the same session hold a reference to the same ts list.